tests/system/marathon_common_tests.py

```diff
@@ -1241,3 +1243,174 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
         "The number of healthy tasks is {}, but {} was expected".format(app['tasksHealthy'], target_instances_count)
 
     client.remove_app(app['id'], True)
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+def test_faultdomains_default():
+    """Tests if the applications that do not have a fault domain specified are launched on the same
+       fault domain as the master.
+    """
+
+    # Discover which is the "local" cluster domain
+    local_domain = get_cluster_local_domain()
+
+    # Get how many agents do we have in the local domain (zone and region) and try to start
+    # that many applications. Mesos should round-robbin the instances in the region
+    region_agents = count_agents_in_faultdomains(regions=local_domain.region)
+
+    # Create an app with default configuration
+    app_def = apps.faultdomain_app(instances=region_agents, suffix="-defaults")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    shakedown.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region (it's ok if they belong to different zones)
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(local_domain.region,))
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+def test_faultdomains_region_only():
+    """Tests if the applications that only have a `region` specified are launched on all the different
+       zones available in that region on the cluster
+    """
+
+    # Find out the biggest region in the cluster
+    (region, zones) = get_biggest_cluster_region()
+
+    assert len(zones) > 1, "The tests require at least 2 zones in the biggest region ({})".format(
```

Comment:
Use a `pytest.mark.skipif` instead.

Reply:
I thought of it, but I have a feeling we are hiding something if I just silently skip this test.
That's because we are testing a crucial feature, and if the cluster is not able to run this kind of test then we should *not* ignore it, but fail proud and loud.

Hence I don't think that this is a classic `skipif` case. For instance, consider the other cases:
* `skipif("shakedown.ee_version() is None")` : We run the tests in a matrix of cluster configurations, and this particular test is testing only the EE feature. Totally legit use.
* `skipif("shakedown.ee_version() == 'strict'")` : Like before, we are running the test in the very specific case of "strict" clusters. Again, totally legit use.

But consider this:
* `skipif("len(get_biggest_cluster_region()[1]) < 2")` : This means that this test cannot run if we hare less than 2 regions in the cluster. But do we have such cases? Because after we merge this PR, *all* of our test matrix is going to run on a 2-region cluster. 

And consider the case we _do_ use `skipif`, but in the future we decide to reduce the cluster size due to some other reason. You are never going to figure out that these tests never run...

I don't know. Both cases have it's positive and negative aspects. I am open on both solutions, I just need some justification, to ease my mind 😄

Reply:
That's actually a good point. The test should fail if we accidentally change the environment. I'd argue it it a prerequisite, though. What do you think about running a check before as we do for Marathon being reachable?

We could have something like
```
def multiple_zones(test_func):
    def wrapped():
         assert len(zones) > 1, "The tests ..."
         return test_func()

    return wrapped

@require(multiple_zones())
def test_faultdomains_default():
    ....
```

Reply:
That's actually a nice approach to it. Let me give it a try.